code/modules/projectiles/projectile/special.dm

```diff
@@ -249,7 +249,7 @@
 
 /obj/item/projectile/energy/teleport/on_hit(atom/target, blocked = 0)
 	if(isliving(target))
-		if(teleport_target)
+		if(get_turf(teleport_target) == target.z || teleport_target.emagged)
```

Comment:
one goof here
you're trying to compare a `/turf` to a z-level value
```dm
get_turf(teleport_target) == target.z
```
you may want to first cast it to a turf var and then check the Z
```dm
var/turf/target_turf = get_turf(teleport_target)
if(target_turf.z == target.z || teleport_target.emagged)
```

Reply:
You commented literally as I was about to, amazing :P

Reply:
we both decided to get on the review grind at the same time lmao

Reply:
I think i forgot to go back and fix this when Qwerty also told me of it, but it worked so it fell by the wayside. Should be fixed now.